drivers/serial/uart_mcux_lpuart.c

```diff
@@ -1208,11 +1244,22 @@ static int mcux_lpuart_line_ctrl_set(const struct device *dev,
 
 	switch (ctrl) {
 	case UART_LINE_CTRL_RTS:
+#ifdef LPUART_HAS_MODEM
 		/* Disable Transmitter and Receiver */
 		config->base->CTRL &= ~(LPUART_CTRL_TE_MASK | LPUART_CTRL_RE_MASK);
 
 		mcux_lpuart_line_ctrl_set_rts(config, val);
+#else
+		ret = -ENOTSUP;
+#endif
+		break;
 
+	case UART_LINE_CTRL_DTR:
+#if defined(FSL_FEATURE_LPUART_HAS_MCR) && FSL_FEATURE_LPUART_HAS_MCR
+		mcux_lpuart_set_dtr(config, val);
+#else
+		ret = -ENOTSUP;
+#endif
```

Comment:
same suggestion about making an initial check instead of doing the same #ifdefs over and over